### PR TITLE
Get user's home directory and use it as cwd and HOME env var value

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

File: shell/shell.go [hunk 1]
```diff
@@ -47,7 +48,12 @@ func ExecuteShell(uid uint32,
 		cmd.SysProcAttr.Credential = &syscall.Credential{Uid: uid, Gid: gid}
 	}
 
+	if _, err := os.Stat(homeDir); !os.IsNotExist(err) {
+		cmd.Dir = homeDir
+	}
+	cmd.Env = append(cmd.Env, fmt.Sprintf("HOME=%s", homeDir))
```
Comment (kacf) [inline shell/shell.go:54]: Is it worth considering using something like `su -l <USER>` to make it a true login shell? I'm not sure exactly what all the differences are, but I know `su` does some extra things, like setting up the environment and putting the shell in its own process group. And specifically related to this PR, it sets the home directory.

Comment (tranchitella) [reply shell/shell.go:54]: @kacf can we expect to have `su -l user` everywhere, in all yocto environment et al?

Comment (tranchitella) [reply shell/shell.go:54]: Great, thanks for the confirmation.

Comment (tranchitella) [reply shell/shell.go:54]: @kacf one thing, though: if we do it this way, we have to run mender-connect as root, don't we?

File: app/daemon.go [hunk 2]
```diff
@@ -372,6 +375,10 @@ func (d *MenderShellDaemon) Run() error {
 	log.Debug("mender-connect connecting to dbus")
 	//dbus main loop, required.
 	dbusAPI, err := dbus.GetDBusAPI()
+	if err != nil {
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply app/daemon.go:380]: Done: https://github.com/mendersoftware/mender-connect/pull/6/commits/5e04d2991be66cded5d4b1cff53c0d2a46535b13

File: shell/shell.go (hunk 1 above)
Comment (kacf) [reply shell/shell.go:54]: You're right. I thought it was possible to run it password-less if you su to the same user as the one running mender-connect, but this doesn't work. In that case I agree, better to stick to what we have for now.

**Review by kacf — APPROVED** [review]

**Pull request merged**